editor/modes/visual-editor/block.js

```diff
@@ -270,7 +313,81 @@ class VisualEditorBlock extends Component {
 
 	onKeyUp( event ) {
 		this.removeOrDeselect( event );
-		this.handleArrowKey( event );
+		//this.handleArrowKey( event );
+	}
+
+	handleToolbarTabCycle( event ) {
+		const { keyCode, shiftKey, target } = event;
+		if ( keyCode !== TAB || ! this.node ) {
+			return;
+		}
+		const block = findDOMNode( this.node );
+		const isVisible = ( elem ) => elem && ! ( elem.offsetWidth <= 0 || elem.offsetHeight <= 0 );
```

Comment:
It's a quick check to see if the element or any ancestors are hidden. It seems that sometimes React hides things rather than re-rendering and actually removing them (at least on Firefox).